packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -607,3 +609,387 @@ class _DefaultProcessUtils implements ProcessUtils {
+    bool runInShell = false,
+    ProcessStartMode mode = ProcessStartMode.normal,
+  }) {
+    return Process.start(
+      sanitizeExecutablePath(_getExecutable(
+        command,
+        workingDirectory,
+        runInShell,
+      ), platform: _platform),
+      _getArguments(command),
+      workingDirectory: workingDirectory,
+      environment: environment,
+      includeParentEnvironment: includeParentEnvironment,
+      runInShell: runInShell,
+      mode: mode,
+    );
+  }
+
+  String _getExecutable(
+    List<String> command,
+    String workingDirectory,
+    bool runInShell,
+  ) {
+    final String commandName = command.first.toString();
+    if (runInShell) {
+      return commandName;
+    }
+    final String executable = getExecutablePath(commandName, workingDirectory, platform: _platform, fileSystem: _fileSystem);
+    if (executable == null) {
+      throw ArgumentError('Could not resolve $commandName to executablePath in $workingDirectory');
+    }
+    return executable;
+  }
+
+  List<String> _getArguments(
+    List<String> command,
+  ) {
+    return command.skip(1).toList();
+  }
+}
+
+/// Sanatizes the executable path on Windows.
+/// https://github.com/dart-lang/sdk/issues/37751
+String sanitizeExecutablePath(String executable, {@required Platform platform }) {
+  if (executable.isEmpty) {
+    return executable;
+  }
+  if (!platform.isWindows) {
+    return executable;
+  }
+  if (executable.contains(' ') && !executable.contains('"')) {
+    // Use quoted strings to indicate where the file name ends and the arguments begin;
+    // otherwise, the file name is ambiguous.
+    return '"$executable"';
+  }
+  return executable;
+}
+
+/// Searches the `PATH` for the executable that [command] is supposed to launch.
+///
+/// This first builds a list of candidate paths where the executable may reside.
+/// If [command] is already an absolute path, then the `PATH` environment
+/// variable will not be consulted, and the specified absolute path will be the
+/// only candidate that is considered.
+///
+/// Once the list of candidate paths has been constructed, this will pick the
+/// first such path that represents an existent file.
+///
+/// Return `null` if there were no viable candidates, meaning the executable
+/// could not be found.
+///
+/// If [platform] is not specified, it will default to the current platform.
+@visibleForTesting
+String getExecutablePath(
+  String command,
+  String workingDirectory, {
+  @required Platform platform,
+  @required FileSystem fileSystem,
+}) {
+  try {
+    workingDirectory ??= fileSystem.currentDirectory.path;
+  } on FileSystemException {
+    // The `currentDirectory` getter can throw a FileSystemException for example
+    // when the process doesn't have read/list permissions in each component of
+    // the cwd path. In this case, fall back on '.'.
+    workingDirectory ??= '.';
+  }
+
+  // TODO(goderbauer): refactor when github.com/google/platform.dart/issues/2
```

Comment:
Not using `package:platform` anymore in the tool. Not sure how that affects this code.

Reply:
Removed